maintenance/wikia/HttpsMigration/migrateCustomCss.php

```diff
@@ -0,0 +1,329 @@
+			preg_match( '/^(www\.)?slot\d*[\.-]images\.wikia\.nocookie\.net$/', $host ) ||
+			$host === 'images.wikia.com' ||
+			$host === 'static.wikia.com' );
+	}
+
+	/**
+	 * Replaces http protocl with https. Protocol-relative urls are not affected.
+	 */
+	private function upgradeToHttps( $url ) {
+		if ( $this->isHttpUrl( $url ) ) {
+			$url = http_build_url( $url, [ 'scheme' => 'https' ] );
+		}
+		return $url;
+	}
+
+	private function forceWikiaImageOverHttps( $url ) {
+		$host = parse_url( $url, PHP_URL_HOST );
+
+		$domain = ( strpos( $host, 'vignette' ) !== FALSE ) ?  'vignette.wikia.nocookie.net' : 'images.wikia.nocookie.net';
+
+		return http_build_url( $url, [ 'scheme' => 'https', 'host' => $domain ] );
+	}
+
+	private $currentHost = null;
+
+	/**
+	 * True if the url point to currently processed wiki.
+	 */
+	private function isCurrentWikiUrl( $url ) {
+		global $wgServer;
+		if ( !$this->currentHost ) {
+			$this->currentHost = parse_url( $wgServer, PHP_URL_HOST );
+		}
+		return parse_url( $url, PHP_URL_HOST ) === $this->currentHost;
+	}
+
+	/**
+	 * Removes the host part of the url leaving only the path.
+	 */
+	private function stripProtocolAndHost( $url ) {
+		$parse_url = parse_url( $url );
+		return $parse_url[ 'path' ] .
+			( ( isset( $parse_url[ 'query' ] ) ) ? '?' . $parse_url[ 'query' ] : '' ) .
+			( ( isset( $parse_url[ 'fragment' ] ) ) ? '#' . $parse_url[ 'fragment' ] : '' );
+	}
+
+	/**
+	 * Fixes wiki urls according to our guidelines:
+	 * - urls to current wiki should be local, without protocol or host
+	 * - urls to other wiki should be protocol-relative.
+	 */
+	private function fixWikiUrl( $url ) {
+		if ( $this->isCurrentWikiUrl( $url ) ) {
+			$url = $this->stripProtocolAndHost( $url );
+		} else {
+			if ( $this->isWikiaComSubdomainUrl( $url ) && $this->isHttpUrl( $url ) ) {
+				$url = wfProtocolUrlToRelative( $url );
+			}
+		}
+		return $url;
+	}
+
+	/**
+	 * For host that we know for sure that support https, upgrade the url to use it.
+	 */
+	private function upgradeThirdPartyUrl( $url ) {
+		$known_https_hosts = [ 'en.wikipedia.org', 'i.imgur.com', 'upload.wikimedia.org', 'fonts.googleapis.com',
+			'commons.wikimedia.org' ];
+		$host = parse_url( $url, PHP_URL_HOST );
+		if ( in_array( $host, $known_https_hosts ) ) {
+			$newUrl = $this->upgradeToHttps( $url );
+			$this->logUrlChange( 'Upgraded third party url', $url, $newUrl );
+			$url = $newUrl;
+		} else {
+			//$this->output( "Found unrecognized third party http url {$url}\n" );
+		}
+		return $url;
+	}
+
+	/**
+	 * Try to make a single url https-ready
+	 */
+	public function fixUrl( $originalUrl ) {
+		$url = $originalUrl;
+
+		if ( $this->isLocalUrl( $url ) ) {
+			//$this->output( "Skipping local link {$url}\n" );
+		} elseif ( $this->isThirdPartyUrl( $url ) ) {
+			if ( $this->isHttpUrl( $url ) ) {
+				$url = $this->upgradeThirdPartyUrl( $url );
+			}
+		} elseif ( $this->isWikiaImageUrl( $url ) ) {
```

Comment:
Do we want to change image URLs that are already HTTPS? Might save unnecessary edit attempts to only focus on HTTP URLs on the first run rather than converting them all to nocookie as well for example. What do you think?

Reply:
Actually it is probably best to change all the instances of images.wikia.com to nocookie.net URLs, so probably nevermind. ;)

Reply:
yes, the goal is to unify image hosts